ubuntu/create_image.sh

```diff
@@ -67,24 +71,23 @@ fi
 tmpdir=$(mktemp -d)
 
 # download iso image
-if [[ $ISO_PATH == "" ]]; then
-    ISO_PATH="ubuntu.iso"
-    echo "Downloading Ubuntu Desktop 22.04.2 image..."
+if [[ ! -f "$ISO_PATH" ]]; then
+    echo "Downloading Ubuntu Desktop ${UBUNTU_VERSION} image..."
     wget -O "$ISO_PATH" $ISO_DOWNLOAD_LINK
 fi
 
 # extract iso contents
 if [[ $ISO_EXTR_PATH == "" ]]; then
     echo "Extracting iso contents..."
     ISO_EXTR_PATH="$tmpdir/extracted"
-    xorriso -osirrox on -indev "$ISO_PATH" -extract / "$ISO_EXTR_PATH" &>/dev/null
+    xorriso -osirrox on -indev "$ISO_PATH" -extract / "$ISO_EXTR_PATH" #&>/dev/null
     # make iso contents modifiable
     chmod -R u+w "$ISO_EXTR_PATH"
 fi
 
-# set up kernel to use preseed
-sed -i -e 's,file=/cdrom/preseed/ubuntu.seed maybe-ubiquity quiet splash,file=/cdrom/preseed/ubuntu.seed iso-scan/filename=${iso_path} auto=true priority=critical boot=casper automatic-ubiquity quiet splash noprompt noshell,g' "$ISO_EXTR_PATH/boot/grub/grub.cfg"
-sed -i -e 's,file=/cdrom/preseed/ubuntu.seed maybe-ubiquity iso-scan/filename=${iso_path} quiet splash,file=/cdrom/preseed/ubuntu.seed iso-scan/filename=${iso_path} auto=true priority=critical boot=casper automatic-ubiquity quiet splash noprompt noshell,g' "$ISO_EXTR_PATH/boot/grub/loopback.cfg"
+# set up kernel to use preseed and serial port
+sed -i -e 's@file=/cdrom/preseed/ubuntu.seed maybe-ubiquity quiet splash@file=/cdrom/preseed/ubuntu.seed iso-scan/filename=${iso_path} auto=true priority=critical boot=casper automatic-ubiquity splash noprompt noshell console=tty0 console=ttyS0,115200n8@g' "$ISO_EXTR_PATH/boot/grub/grub.cfg"
+sed -i -e 's@file=/cdrom/preseed/ubuntu.seed maybe-ubiquity iso-scan/filename=${iso_path} quiet splash@file=/cdrom/preseed/ubuntu.seed iso-scan/filename=${iso_path} auto=true priority=critical boot=casper automatic-ubiquity splash noprompt noshell console=tty0 console=ttyS0,115200n8@g' "$ISO_EXTR_PATH/boot/grub/loopback.cfg"
```

Comment:
@mkopec, @TomaszAIR, so, should serial be enabled in installer? According to comment https://github.com/Dasharo/preseeds/pull/11#issuecomment-2003824665 there is no reason to do so.

Reply:
It may make it easier to debug issues, so if it doesn't cause problems I'd leave it enabled

Reply:
> It may make it easier to debug issues

Agree.

Reply:
Definitely it does not hurt to keep it.